rust/agents/relayer/src/msg/processor.rs

```diff
@@ -63,92 +66,106 @@ impl MessageProcessor {
         }
     }
 
+    /// Tries to get the next message to process.
+    ///
+    /// If no message with self.message_nonce is found, returns None.
+    /// If the message with self.message_nonce is found and has previously
+    /// been marked as processed, increments self.message_nonce and returns
+    /// None.
+    fn try_get_unprocessed_message(&mut self) -> Result<Option<HyperlaneMessage>, DbError> {
+        // First, see if we can find the message so we can update the gauge.
+        if let Some(message) = self.db.message_by_nonce(self.message_nonce)? {
+            // Update the latest nonce gauge if the message is destined for one
+            // of the domains we service.
+            if let Some(metrics) = self.metrics.get(message.destination) {
+                metrics.set(message.nonce as i64);
+            }
+
+            // If this message has already been processed, on to the next one.
+            if self
+                .db
+                .retrieve_message_processed(self.message_nonce)?
+                .is_some()
+            {
+                debug!(
+                    msg_nonce=?self.message_nonce,
+                    "Message already marked as processed in DB");
+                self.message_nonce += 1;
+                return Ok(None);
+            }
+
+            return Ok(Some(message));
```

Comment:
Nit:
```
if <cond> {
 ...
 Ok(None)
} else {
 Ok(Some(message))
}
```

(more clear IMO when all paths have a result rather than making it look like there are break-like returns to halt fn exec)